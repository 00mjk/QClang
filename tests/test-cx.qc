int main() {
	qubit q1;
	qubit q2;

	/* entangle q1 and q2 */
	(q1, q2) = CX(hadamard(false), false);
	/* unentangle */
	(q1, q2) = CX(q1, q2);
	q1 = hadamard(q1);
	measure(q1);
	measure(q2);
	return 0;
}
