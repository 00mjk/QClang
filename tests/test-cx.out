OPENQASM 2.0;
include "qelib1.inc";
qreg q[1];
creg c[1];
h q;
qreg temp_0[1];
h temp_0;
qreg temp_1[1];
cx temp_0, temp_1;
cx temp_0, temp_1;
h temp_0;
creg temp_0_mb2[1];
measure temp_0 -> temp_0_mb2;
creg temp_1_mb3[1];
measure temp_1 -> temp_1_mb3;
